efibootmgr: make modifying of boot records optional, disabled by default.

CMB opened the pull request:
Also changed the name of the config file, since the syntax
is much different.

See issue #5165 for the discussion.

aeosynth commented on srcpkgs/efibootmgr/INSTALL.msg:
```diff
@@ -1,11 +1,12 @@
-From now on, efibootmgr automatically generates efi boot entries to
+efibootmgr can automatically generate EFI boot entries to boot
 directly boot the kernel after every kernel update or installation.
```
> to boot directly boot

the added 'boot' is unnecessary

Vaelatern commented on srcpkgs/efibootmgr/template:
```diff
@@ -23,7 +23,7 @@ do_install() {
 	vlicense COPYING
 	vmkdir etc/
-	touch ${DESTDIR}/etc/efibootmgr-options.conf
+	vinstall "${FILESDIR}/efibootmgr-kernel-hook.conf" 644 etc/
```
Please use `vconf`

Vaelatern commented on srcpkgs/efibootmgr/files/efibootmgr-kernel-hook.conf:
```diff
@@ -0,0 +1,6 @@
+# Options for the kernel hook script installed by the efibootmgr package.
+modify_efi_entries=0
```
Please change these options to be all-caps.

CMB commented:
Fixed to use vconf.
Options in the config file are all-caps.
Also added braces around variable names in a couple places.
Typo in the install message is fixed.

Thanks for the reviews.

Vaelatern commented:
@Duncaen Should the conf file be moved? To `/etc/default/` seems reasonable, but I am not sure.
the problem is once this gets merged, moving a file like this will be more problematic.
